src/java/org/apache/cassandra/db/streaming/CassandraOutgoingFile.java

```diff
@@ -127,6 +135,8 @@ public String getName()
     @Override
     public long getSize()
     {
+        // File size may not be accurate due to concurrent stats update or index summary redistribution. Since it's only
+        // used for metrics, an estimation should be fine.
         return header.size();
```

Comment:
If this doesn't happen much, what about just generating the header on the fly? We'd at least avoid the slight confusion of having both an instance field and the `current` header for entire-SSTable streaming...

Reply:
I didn't compute size in flight because the size computation is slightly slow for non-entire-streaming with compression..